last_seen.py

```diff
@@ -0,0 +1,30 @@
+from brutal.core.plugin import BotPlugin, cmd, event
+from time import strftime, gmtime
+
+
+class LastSeen(BotPlugin):
+    def setup(self, *args, **kwargs):
+        self.seen = self.open_storage('last_seen')
+
+    @event
+    def seen_evt(self, event):
+        events = ['quit', 'part', 'kick', 'join', 'rename', 'topic', 'message']
+        if event.event_type in events:
+            nick = event.meta['nick']
+            time = gmtime()
+            evt_type = event.event_type
+            self.seen[nick] = (evt_type, time)
+
+    @cmd
+    def last_seen(self, event):
+        args = event.args
+        if len(args) < 1:
+            return "Name argument missing."
+        user = args[0]
+
+        if user not in self.seen:
+            return "This user was never here."
+
+        evt, time = self.seen[user]
+        ret_time = strftime("%H:%M %d.%m.%Y", time)
```

Comment:
@JakubNvk you could reassign the strftime-d value back to time -- it's not used anywhere else anyway.